### DISCUSSION TITLE
[Bug] maven pull pulsar-broker 2.8.4 expose java.lang.StackOverflowError

### DISCUSSION THREAD

**Discussion by Liu-ZhenYu** [apache/pulsar#20895] [Q&A] [answered]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

pulsar 2.8.4
maven 3.6.3

### Minimal reproduce step

maven pom.xml
` 
<dependencies>
    <dependency>
      <groupId>org.apache.pulsar</groupId>
      <artifactId>pulsar-broker</artifactId>
      <version>2.8.4</version>
    </dependency>
</dependencies>
`

### What did you expect to see?

maven pull pulsar-broker 2.8.4 success

### What did you see instead?

maven pull pulsar-broker 2.8.4 success

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by liangyepianzhou** [comment]
@BewareMyPower, Could you please take a look?

**Comment by BewareMyPower** [comment]
It's a strange error, I figured out a fix by importing the BK stream storage, could you try it?

```xml
    <dependency>
      <groupId>org.apache.bookkeeper</groupId>
      <artifactId>stream-storage-java-client</artifactId>
      <version>4.15.4</version>
    </dependency>
```

**Comment by Liu-ZhenYu** [comment, marked as answer]
@BewareMyPower 
The problem can be solved in your way, and I can solve it in another way
```java
<dependency>
      <groupId>org.apache.pulsar</groupId>
      <artifactId>pulsar-broker</artifactId>
      <version>2.8.4</version>
      <exclusions>
        <exclusion>
          <groupId>io.grpc</groupId>
          <artifactId>grpc-all</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
```
